src/foam/nanos/u2/navigation/SideNavigationItemView.js

```diff
@@ -1,45 +1,74 @@
 foam.CLASS({
-    package: 'foam.nanos.u2.navigation',
-    name: 'SideNavigationItemView',
-    extends: 'foam.u2.View',
-  
-    properties: [
-      {
-        class: 'Boolean',
-        name: 'expanded',
-        value: true
-      }
-    ],
-  
-    methods: [
-      function initE() {
-        var view = this;
-  
-        this
-          .start('a')
-            .add(this.data.label)
-            .on('click', function() {
-              var menu = view.data;
-              menu.children.select().then(function(c) {
-                if ( c.array.length === 0 ) {
-                  menu.launch(view.__context__, view);
-                } else {
-                  view.expanded = ! view.expanded;
-                }
+  package: 'foam.nanos.u2.navigation',
+  name: 'SideNavigationItemView',
+  extends: 'foam.u2.View',
+
+  imports: [
+    'currentMenu',
+  ],
+
+  requires: [
+    'foam.nanos.menu.Menu',
+  ],
+
+  css:`
+    ^selected {
+      opacity:1 !important;
+      text-shadow: 0 0 0px white, 0 0 0px white;
+    }
+  `,
+
+  properties: [
+    {
+      class: 'Boolean',
+      name: 'expanded',
+      value: true
+    },
+    {
+      class: 'Int',
+      name: 'paddingLeft'
+    },
+    {
+      class: 'Int',
+      name: 'fontSize'
+    },
+    {
+      class: 'Float',
+      name: 'opacity'
+    }
+  ],
+
+  methods: [
+    function initE() {
+      var view = this;
+
+      this
+        .addClass(this.myClass())
+        .start('a')
+          .add(this.data.label)
+          .enableClass(this.myClass('selected'), view.currentMenu$.map(function (value) {
+            return window.location.hash.substring(1) === (view.data.id);
+          }))
+          .style({'padding-left': view.paddingLeft + 'px', 'font-size': view.fontSize + 'px', 'opacity': view.opacity})
+          .on('click', function() {
+            var menu = view.data;
+            menu.children.select().then(function(c) {
```

Comment:
I don't think this is required. Just always call launch() and set expanded.

Reply:
menu.launch() doesn't seem to work when c.array.length !== 0 (when menu is a submenu that has children). because the launch method in SubMenu.js calls the open method in PopupMenu.js, which doesn't work here. 
Should I fix launch, so that it works for a submenu in the side bar (instead of just the ones in top navigation), or should I just leave the if condition?

Reply:
Updated code is in PR: https://github.com/foam-framework/foam2/pull/1634